examples/core/Graphormer/train.py

```diff
@@ -0,0 +1,483 @@
+            loss,
+            attn_mask,
+            node_feat,
+            in_degree,
+            out_degree,
+            path_data,
+            dist,
+        )
+        th.cuda.empty_cache()
+
+    epoch_loss /= iter + 1
+
+    evaluator = Evaluator(name="ogbg-molhiv")
+    epoch_train_metric = evaluator.eval(
+        {"y_pred": th.cat(list_scores), "y_true": th.cat(list_labels)}
+    )["rocauc"]
+
+    return epoch_loss, epoch_train_metric
+
+
+def evaluate_network(model, data_loader):
+    model.eval()
+    epoch_test_loss = 0
+    loss_fn = nn.BCEWithLogitsLoss()
+    with th.no_grad():
+        list_scores = []
+        list_labels = []
+        for iter, (
+            batch_labels,
+            attn_mask,
+            node_feat,
+            in_degree,
+            out_degree,
+            path_data,
+            dist,
+        ) in enumerate(data_loader):
+            device = accelerator.device
+
+            batch_scores = model(
+                node_feat.to(device),
+                in_degree.to(device),
+                out_degree.to(device),
+                path_data.to(device),
+                dist.to(device),
+                attn_mask=attn_mask,
+            )
+
+            # Gather all predictions and targets
+            all_predictions, all_targets = accelerator.gather_for_metrics(
+                (batch_scores, batch_labels)
+            )
+            loss = loss_fn(all_predictions, all_targets.float())
+
+            epoch_test_loss += loss.item()
+            list_scores.append(all_predictions)
+            list_labels.append(all_targets)
+
+        epoch_test_loss /= iter + 1
+
+        evaluator = Evaluator(name="ogbg-molhiv")
+        epoch_test_metric = evaluator.eval(
+            {"y_pred": th.cat(list_scores), "y_true": th.cat(list_labels)}
+        )["rocauc"]
+
+    return epoch_test_loss, epoch_test_metric
+
+
+def train_val_pipeline(params):
+
+    dataset = MolHIVDataset()
+
+    train_loader = GraphDataLoader(
+        dataset.train,
+        batch_size=params.batch_size,
+        shuffle=True,
+        collate_fn=dataset.collate,
+        pin_memory=True,
+        num_workers=16,
+    )
+    val_loader = GraphDataLoader(
+        dataset.val,
+        batch_size=params.batch_size,
+        shuffle=False,
+        collate_fn=dataset.collate,
+        pin_memory=True,
+        num_workers=16,
+    )
+    test_loader = GraphDataLoader(
+        dataset.test,
+        batch_size=params.batch_size,
+        shuffle=False,
+        collate_fn=dataset.collate,
+        pin_memory=True,
+        num_workers=16,
+    )
+
+    # load pretrain model
+    download(url="https://data.dgl.ai/pre_trained/graphormer_pcqm.pth")
+    model = Graphormer()
+    state_dict = th.load("graphormer_pcqm.pth")
+    model.load_state_dict(state_dict)
+    # reset output layer parameters
+    model.reset_output_layer_parameters()
+    num_epochs = 16
+    tot_updates = 33000 * num_epochs / params.batch_size
```

Comment:
Where does `33000` come from?

Reply:
Approximate number of training examples, the same as the author's implementation.